src/container/oid_iv.c

```diff
@@ -130,7 +131,14 @@ oid_iv_ent_update(struct ds_iv_entry *ns_entry, struct ds_iv_key *iv_key,
 	D_ASSERT(priv != NULL);
 
 	entry = ns_entry->iv_value.sg_iovs[0].iov_buf;
-	ABT_mutex_lock(entry->lock);
+	rc = ABT_mutex_trylock(entry->lock);
+	/* For retry requests, from _iv_op(), the lock may not be released
+	 * in some cases.
```

Comment:
"some cases" for example the oid_iv_ent_update() hold the lock and return DER_IVCB_FORWARD, but expected oid_iv_ent_refresh() is not executed to release the lock? (for example timeout and retry, or pm ver/IV_tree changed).

maybe better to don't hold lock between network operation, if possible.

Reply:
>but expected oid_iv_ent_refresh() is not executed to release the lock? (for example timeout and retry, or pm ver/IV_tree changed).

If the RPC to its parent is sent, then refresh will be called in any case.  But if the failure happened before RPC is sent, for example get_parent() failed by -1036, then refresh will not be called before another retry. that is why it caused the deadlock here.

yeah, removing this lock(serialization) might need some complicate change. Yes, you can replace the lock with the flag etc, but no difference if we change this to trylock, since it is not recursive lock anyway.